core/lib/pattern_exporter.js

```diff
@@ -14,18 +15,50 @@ var pattern_exporter = function () {
   function exportPatterns(patternlab) {
     //read the config export options
     var exportPartials = patternlab.config.patternExportPatternPartials;
+    var exportAll = patternlab.config.patternExportAll;
+
+    if (exportAll) {
+      for (var i = 0; i < patternlab.patterns.length; i++) {
+        if (!patternlab.patterns[i].patternPartial.startsWith('-')) {
+          exportSinglePattern(patternlab, patternlab.patterns[i]);
```

Comment:
I'd suggest adding a check for to make sure `patternlab.patterns[i].isPattern` === `true` before calling `exportSinglePattern()` to avoid having the meta head and meta foot from being exported.